### DISCUSSION TITLE
Non-baseline / 42e01f H264 profiles for decoding

### DISCUSSION THREAD

**Discussion by alexthedamager** [OvenMediaLabs/OvenMediaEngine#1270] [Help]
I'm running some tests using ffmpeg to stream h264 10-bit video with profile high422 to OvenMediaEngine via RTMP:

The stream plays in Chrome, Safari and Firefox, but in webrtc-internals the stream details are: H264 (98, level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42e01f)

Notably `profile-level-id=42e01f` - `42e01f` is baseline profile. In OME source code, there are a number of references to 42e01f including in `h264_converter.h`

Does anyone know what is going on here - is OME receiving a 10-bit high422 h264 stream but telling the browser to play it back with baseline profile? I am running OME without video transcoding enabled - pass-through only.

Does the SDP need to be modified so that `profile-level-id=` matches the encoder h264 profile?

**Comment by getroot** [comment]
I thought a lot about it a long time ago and eventually fixed the Codec Profile in the WebRTC SDP temporarily so that it does not change. Because at the time it was thought to be more conducive to stability. See comments below.

https://github.com/AirenSoft/OvenMediaEngine/blob/master/src/projects/publishers/webrtc/rtc_stream.cpp#L560

Delete the code below and it will work as you intended. After that, it would be very helpful if you share your test result (browser playback result according to H.264 profile) to the community.

`profile_string = H264_CONVERTER_DEFAULT_PROFILE;`
